### PR TITLE
Added a Title to the JFXDecorator (see Main Demo)

### PR DESCRIPTION
**Pull request by jlarrieux**
Added a Title to the JFXDecorator. This title can be set by calling setTitle(String) function on the JFXDecorator. Created a jfx-decorator-label css for styling. Please feel free to change that styling. Also this is my 1st contribution to any open source project. Please provide feedback

### CODE REVIEW

**Comment by jfoenixadmin** [comment]
Thanks for your contribution 👍 , it's looks fine. However was thinking, maybe replacing the title property with graphic property would make it more general for the user.

**Comment by jlarrieux** [comment]
I can add that as well. I believe having a simple "setTitle()" method for JFXDecorator is very valuable. I will add graphics in addition to that.

Also I wanted to refactor the JFXDecorator constructor that takes in Stage, Node, and 3 boolean values. This constructor starts at line 109 and ends at line 436. I was planning on breaking it down into smaller private functions for the sake of clarity and clean code. Just wanted to get approval first before doing it. If approved, I can work on both of these items.

**Comment by jfoenixadmin** [comment]
Sounds great 👍

File: demo/src/main/resources/css/jfoenix-main-demo.css [hunk 1]
```diff
@@ -771,6 +771,12 @@
     -fx-border-width: 0 4 4 4;
 }
 
+.jfx-decorator-label {
```
Comment (skmedix) [inline demo/src/main/resources/css/jfoenix-main-demo.css]: Other components does have selectors like `.jfx-decorator .label`

**Comment by jfoenixadmin** [comment]
@jlarrieux did you finish your changes to I can merge your pull request with master branch?

**Comment by jlarrieux** [comment]
Yes. Please let me know if you want me to do anything else on this.

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 2]
```diff
@@ -80,6 +81,9 @@
     protected JFXButton btnFull;
     protected JFXButton btnClose;
     protected JFXButton btnMin;
+    protected Label title;
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 3]
```diff
@@ -80,6 +81,9 @@
...
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 4]
```diff
@@ -115,6 +119,80 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
         setPickOnBounds(false);
         this.getStyleClass().add("jfx-decorator");
 
+        initializeButtons();
+        initializeContainers(node,fullScreen, max, min);
+
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 5]
```diff
@@ -115,6 +119,80 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
...
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 6]
```diff
@@ -115,6 +119,80 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
...
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 7]
```diff
@@ -115,6 +119,80 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
...
+        addEventFilter(MouseEvent.MOUSE_RELEASED, (mouseEvent) -> isDragging = false);
+
+        this.setOnMouseDragged((mouseEvent) -> handleDragEventOnDecoratorPane(mouseEvent));
+    }
+
+
+
+
+    private void initializeButtons(){
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 8]
```diff
@@ -115,6 +119,80 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
...
+        // show the drag cursor on the borders
+        addEventFilter(MouseEvent.MOUSE_MOVED, (mouseEvent) -> showDragCursorOnTheborders(mouseEvent));
+
+
+        // handle drag events on the decorator pane
+        addEventFilter(MouseEvent.MOUSE_RELEASED, (mouseEvent) -> isDragging = false);
+
+        this.setOnMouseDragged((mouseEvent) -> handleDragEventOnDecoratorPane(mouseEvent));
+    }
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 9]
```diff
@@ -115,6 +119,80 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
...
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 10]
```diff
@@ -115,6 +119,80 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
...
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Useless linebreak

File: jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java [hunk 11]
```diff
@@ -268,171 +363,120 @@ public JFXDecorator(Stage stage, Node node, boolean fullScreen, boolean max, boo
...
+           } else if (isTopEdge(x, y, boundsInParent)) {
+               this.setCursor(Cursor.N_RESIZE);
+           } else if (isBottomEdge(x, y, boundsInParent)) {
+               this.setCursor(Cursor.S_RESIZE);
+           } else {
+               this.setCursor(Cursor.DEFAULT);
+           }
+       }
+   }
+
```
Comment (skmedix) [inline jfoenix/src/main/java/com/jfoenix/controls/JFXDecorator.java]: Again, why you're using that many linebreaks?

**Comment by jlarrieux** [comment]
Just visually more pleasing (add a break to different sections) but I can take them off

**Comment by skmedix** [comment]
Would be great, because at the moment it doesn't fit with the rest of the code

**Comment by jfoenixadmin** [comment]
One more thing @jlarrieux , I noticed that you didn't add the graphic property yet.

**Comment by jlarrieux** [comment]
To be honest, I am stuck with the graphic. Should I do SVG or Image from JavaFX? Both these options take arguments in the constructors.
I was planning to create a blank and transparent image (maybe even zero size) so that it only shows if an image has been specifically declared. This image would be in the same container as the title label. Not sure how to accomplish this.  Any suggestion

**Comment by jfoenixadmin** [comment]
The graphic property should be implemented similar to other JavaFX controls graphic property. So it should be of type ObjectProperty<Node>. You can check JFXToolBar or JFXDialogLayout for a simple example.

**Comment by jfoenixadmin** [comment]
Looks Great!! :)
I'll do some minor refactoring regarding the methods names, just to be consistent with javafx naming and params. Thank you @jlarrieux for contributing to JFoenix 👍

**Pull request merged**